utils/Trackers.h

```diff
@@ -133,6 +133,26 @@ class RateTracker
     uint64_t _prevTimestamp;
 };
 
+template <size_t NUM_BUCKETS, uint64_t BUCKET_SIZE_MS, uint64_t SNAPSHOT_DELTA_MS>
+struct TrackerWithSnapshot
+{
+    TrackerWithSnapshot() : tracker(BUCKET_SIZE_MS), _lastSnapshotUpdatedAt(0) { snapshot.store(0.0); }
+    utils::RateTracker<NUM_BUCKETS> tracker;
+    std::atomic<double> snapshot;
```

Comment:
there is a typedef for this std::atomic_double

Reply:
Are you sure?

Reply:
I mean atomic<double> is not fully implemented, and I can't find atomic_double macro.